### PR TITLE
Core: updated argon2 to latest release

### PR DESCRIPTION
**Pull request by damaex**
**Changes proposed:**

- updated argon2 for arm/aarch64 support

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)
- [x] build
- [x] in game (on raspberry pi 4)

**Known issues and TODO list:** (add/remove lines as needed)

### CODE REVIEW

**Comment by jackpoz** [comment]
Please update https://github.com/TrinityCore/TrinityCore/blob/33aa5d2e39af867e85c14a2c412a23d08c51cc25/dep/PackageList.txt too with the new version of the package (for example the commit hash)

File: dep/argon2/CMakeLists.txt [hunk 1]
```diff
@@ -9,6 +9,15 @@
 # implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 
 file(GLOB_RECURSE sources *.c)
+
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+    list(REMOVE_ITEM sources
```
Comment (Shauren) [inline dep/argon2/CMakeLists.txt]: Please fix indent size here (and in the else part) to match rest of the file

**Comment by damaex** [comment]
The version in the PackageList.txt is already on point with the latest Upstream:
https://github.com/P-H-C/phc-winner-argon2/commit/62358ba2123abd17fccf2a108a301d4b52c01a7c

**Comment by jackpoz** [comment]
does that mean that the version in PackagesList.txt doesn't match the version we have in /dep ?

**Comment by jackpoz** [comment]
@Treeston could you take a look at this PR ? did you remove some code from argon2 when including it in TC ? is it ok to include it back ?

**Comment by Treeston** [comment]
I stripped threading logic because we don't use it, and I didn't want foreign MT code in the TC codebase.

**Comment by Shauren** [comment]
You should have just defined ARGON2_NO_THREADS

**Comment by Treeston** [comment]
...I had some reason for not doing that at the time, though I have to admit I don't recall it.

**Comment by jackpoz** [comment]
@Shauren were should ARGON2_NO_THREADS be defined ? In cmake somewhere ?

**Comment by Shauren** [comment]
in argon2 CMakeLists.txt

**Comment by damaex** [comment]
added ARGON2_NO_THREADS definition to argon2 library

File: dep/argon2/CMakeLists.txt [hunk 2]
```diff
@@ -9,18 +9,28 @@
 # implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 
 file(GLOB_RECURSE sources *.c)
+add_definitions(-DARGON2_NO_THREADS)
```
Comment (Shauren) [inline dep/argon2/CMakeLists.txt]: add_definitions is the old way of specifying compile flags, you should use

```
target_compile_definitions(argon2
  PRIVATE
    -DARGON2_NO_THREADS)
```

File: dep/argon2/CMakeLists.txt [hunk 3]
```diff
@@ -9,18 +9,28 @@
...
+    ${CMAKE_CURRENT_SOURCE_DIR}/argon2/ref.c)
+endif()
+
 add_library(argon2 STATIC
 ${sources})
 
 target_include_directories(argon2
-PUBLIC
+  PUBLIC
   ${CMAKE_CURRENT_SOURCE_DIR})
```
Comment (Shauren) [inline dep/argon2/CMakeLists.txt]: add extra indent level to this line too

File: dep/argon2/CMakeLists.txt [hunk 4]
```diff
@@ -9,18 +9,28 @@
...
 
 target_link_libraries(argon2
-PRIVATE
+  PRIVATE
   trinity-dependency-interface)
 
 set_target_properties(argon2
-PROPERTIES
+  PROPERTIES
   FOLDER
```
Comment (Shauren) [inline dep/argon2/CMakeLists.txt]: ++indent (and on next line too)

File: dep/argon2/CMakeLists.txt [hunk 5]
```diff
@@ -9,18 +9,28 @@
...
 
 target_include_directories(argon2
-PUBLIC
+  PUBLIC
   ${CMAKE_CURRENT_SOURCE_DIR})
 
 target_link_libraries(argon2
-PRIVATE
+  PRIVATE
   trinity-dependency-interface)
```
Comment (Shauren) [inline dep/argon2/CMakeLists.txt]: ++indent

**Review by Shauren — CHANGES_REQUESTED** [review]

**Review by Shauren — APPROVED** [review]
This looks great, ship when it passes build test

**Comment by illfated** [comment]
🔼  :shipit:

**Pull request merged**